tasks/diff.py

```diff
@@ -0,0 +1,189 @@
+
+                            depsfile = os.path.join(tmpdir, f"{target}-{branch_name}")
+                            flavor = details.get("flavor", AgentFlavor.base)
+                            build = details.get("build", binary)
+                            build_tags = get_default_build_tags(
+                                build=build, arch=arch, platform=platform, flavor=flavor
+                            )
+                            env = {"GOOS": goos, "GOARCH": goarch}
+                            ctx.run(f"{dep_cmd} -tags \"{' '.join(build_tags)}\" > {depsfile}", env=env)
+        finally:
+            ctx.run(f"git checkout -q {current_branch}")
+
+        # compute diffs for each target
+        for binary, details in binaries.items():
+            for combo in details.get("platforms"):
+                platform, arch = combo.split("/")
+                goos, goarch = GOOS_MAPPING.get(platform), GOARCH_MAPPING.get(arch)
+                target = f"{binary}-{goos}-{goarch}"
+
+                prdeps = os.path.join(tmpdir, f"{target}-current")
+                maindeps = os.path.join(tmpdir, f"{target}-main")
+                res = ctx.run(
+                    f"diff -u0 {maindeps} {prdeps} | grep -v '^@@' | grep -v '^[+-][+-]'", hide=True, warn=True
+                )
+                if len(res.stdout) > 0:
+                    diffs[target] = res.stdout.strip()
+
+        # output, also to file if requested
+        if len(diffs) > 0:
+            pr_comment = [
+                f"Baseline: {baseline_ref}",
+                f"Comparison: {commit_sha}\n",
+            ]
+            for binary, details in binaries.items():
+                for combo in details.get("platforms"):
+                    platform, arch = combo.split("/")
+                    goos, goarch = GOOS_MAPPING.get(platform), GOARCH_MAPPING.get(arch)
+                    target = f"{binary}-{goos}-{goarch}"
+                    prettytarget = f"{binary} {goos}/{goarch}"
+
+                    if target in diffs:
+                        targetdiffs = diffs[target]
+                        add, remove = patch_summary(targetdiffs)
+                        color_add = color_message(f"+{add}", "green")
+                        color_remove = color_message(f"-{remove}", "red")
+                        print(f"== {prettytarget} {color_add}, {color_remove} ==")
+                        print(f"{color_patch(targetdiffs)}\n")
+
+                        summary = f"<summary>{prettytarget} +{add}, -{remove}</summary>"
+                        diff_block = f"\n```diff\n{targetdiffs}\n```\n"
+                        pr_comment.append(f"<details>{summary}\n{diff_block}</details>\n")
+                    else:
+                        print(f"== {prettytarget} ==\nno changes\n")
+
+            if report_file:
+                with open(report_file, 'w') as f:
+                    f.write("\n".join(pr_comment))
+        else:
```

Comment:
So in this case the pr commenter will create an empty comment? If so, it would be nice if it rather created a comment saying that nothing changed, I think.

Reply:
That would mean every PR would get a comment. I was trying to limit the noise and keep it to only commenting on PRs that had changes.

Reply:
I'm not sure I understand. Are you saying that the PR commenter will only create a comment if the input file is non-empty? Because from what I'm seeing, your changes will always call the commenter, even if it will be passed an empty input file.

Reply:
> Are you saying that the PR commenter will only create a comment if the input file is non-empty?

Correct, and it will actually delete an existing comment. So if there were changes initially, then further commits remove the changes, the comment will be deleted.

Reply:
I have tested this exact behavior.